src/main/java/org/opensearch/neuralsearch/sparse/query/SparseAnnQueryBuilder.java

```diff
@@ -317,4 +329,21 @@ private static Map<String, Float> preprocessQueryTokens(Map<String, Float> token
         return intTokens.entrySet().stream().collect(Collectors.toMap(e -> String.valueOf(e.getKey()), Map.Entry::getValue));
     }
 
+    private static float getQuantizationCeilSearch(QueryShardContext context, String fieldName) {
+        // different field infos should have the same value for quantization ceiling search
+        float quantizationCeilSearch = DEFAULT_QUANTIZATION_CEILING_SEARCH;
+        try {
+            for (LeafReaderContext leafContext : context.searcher().getTopReaderContext().leaves()) {
+                FieldInfos fieldInfos = leafContext.reader().getFieldInfos();
+                FieldInfo fieldInfo = fieldInfos.fieldInfo(fieldName);
+                if (fieldInfo != null) {
+                    quantizationCeilSearch = ByteQuantizationUtil.getCeilingValueSearch(fieldInfo);
+                    break;
+                }
+            }
+        } catch (Exception e) {
+            log.error(String.format(Locale.ROOT, "Failed to get quantization ceiling search value for field [%s]", fieldName), e);
```

Comment:
nit: `log.error("Failed to get quantization ceiling search value for field [{}]", fieldName, e);`

Reply:
I was saying you don't need to use a String.format within a log